src/Akeneo/Platform/Bundle/UIBundle/Resources/public/js/product/field/metric-field.js

```diff
@@ -38,6 +39,7 @@ define([
                 templateContext.i18n = i18n;
                 templateContext.units = measurementFamily.units;
                 templateContext.locale = templateContext.locale || templateContext.context.locale;
+                templateContext.uiLocale = UserContext.get('uiLocale');
```

Comment:
This one was added by me
https://github.com/akeneo/pim-community-dev/commit/f7bf1df71a044d09102de9d304b7986fda800798#diff-560f9f5174f27c976d1dc73643bcb030R40
I believe you can use uiLocale directly:
```suggestion
                templateContext.locale = UserContext.get('uiLocale');
```

Reply:
`locale` is considered as `catalogLocale` and is used for the value and displaying the flag.
If I do as you suggest, it's not working as expected, see:
 ![Screenshot_2020-03-04_12-09-17](https://user-images.githubusercontent.com/1421130/75873887-059f4380-5e11-11ea-811f-1b4dfa423a59.png)

Reply:
So maybe you can just remove line 41 (as it was me who added it)